python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -172,6 +175,37 @@ def resolve_partitionKeys(self, _graphene_info):
                 ]
         return []
 
+    def resolve_latestMaterializationPerPartition(self, _graphene_info, **kwargs):
+        from ..implementation.fetch_assets import get_asset_events
+
+        get_timestamp = lambda event: event.timestamp
+        get_partition = (
+            lambda event: event.dagster_event.step_materialization_data.materialization.partition
+        )
+
+        partitions = kwargs.get("partitions")
+        events_for_partitions = get_asset_events(
+            _graphene_info,
+            self._external_asset_node.asset_key,
+            partitions,
+        )
+
+        latest_materialization_by_partition = {}
+        for event in events_for_partitions:
+            event_partition = get_partition(event)
+            if event_partition not in latest_materialization_by_partition:
+                latest_materialization_by_partition[event_partition] = event
+            else:
+                if get_timestamp(event) > get_timestamp(
+                    latest_materialization_by_partition[event_partition]
+                ):
+                    latest_materialization_by_partition[event_partition] = event
+
+        return [
+            GrapheneAssetMaterialization(event=event)
```

Comment:
Are we just relying on the asset materialization to have the partition information populated on it?  should it be ordered in any way?

Reply:
I modified the method to return back a list with asset materializations in the same order as the partition keys accepted (with `None` if there is no materialization for the partition), so partition information won't need to be fetched from the asset materialization.